lib/linguist/languages.yml

```diff
@@ -2753,6 +2753,17 @@ Meson:
   tm_scope: source.meson
   ace_mode: text
   language_id: 799141244
+Meta Programming System:
+  type: programming
+  color: "#007800"
+  extensions:
+  - ".mps"
+  - ".mpsr"
```

Comment:
Given [the search request you provided](https://github.com/search?utf8=%E2%9C%93&q=extension%3Ampsr+persistence+&type=Code), I don't think there are enough `.mpsr` files to mandate its support in Linguist yet.